ui/components/Password.tsx

```diff
@@ -21,67 +32,90 @@ const Password: FC<Props> = ({
   setValid,
   label = "Password",
 }) => {
-  const [confirmation, setConfirmation] = React.useState("");
-  const [hasPassBlur, setPasswordBlurred] = React.useState(false);
-  const [hasConfirmBlur, setConfirmationBlurred] = React.useState(false);
+  const [confirmation, setConfirmation] = useState("");
+  const [hasTouched, setHasTouched] = useState(false);
+  const [hasBlurred, setHasBlurred] = useState(false);
 
-  const getStatus = (check: PasswordCheckType) => {
-    if (!hasPassBlur) {
-      return "neutral";
-    }
+  const debouncedPassword = useDebounce(password, DEBOUNCE_DURATION);
+  const debouncedConfirmation = useDebounce(confirmation, DEBOUNCE_DURATION);
 
+  const validateCheck = (check: PasswordCheckType, value: string): boolean => {
     switch (check) {
       case "lowercase":
-        return /[a-z]/.test(password) ? "success" : "error";
+        return /[a-z]/.test(value);
       case "uppercase":
-        return /[A-Z]/.test(password) ? "success" : "error";
+        return /[A-Z]/.test(value);
       case "number":
-        return /[0-9]/.test(password) ? "success" : "error";
+        return /\d/.test(value);
       case "length":
-        return password.length >= 8 ? "success" : "error";
+        return value.length >= 8;
     }
   };
```

Comment:
issue(readability): please move this function out to refactor it to a static function, so the body of the component is not made heavy to read